searx/data/useragents.json

```diff
@@ -7,5 +7,5 @@
         "Windows NT 10.0; Win64; x64",
         "X11; Linux x86_64"
     ],
-    "ua": "Mozilla/5.0 ({os}; rv:{version}) Gecko/20100101 Firefox/{version}"
-}
\ No newline at end of file
+    "ua": "Mozilla/5.0 ({os}; rv:109.0) Gecko/20100101 Firefox/{version}"
```

Comment:
That's not how it should be done.

We randomly send the requests between the two versions described in https://github.com/searxng/searxng/blob/master/searx/data/useragents.json#L2

We are not going to change that behavior.

Reply:
maybe @dalf know a bit more about this process

Reply:
We still send the version randomly, look at the end of the User-Agent. This matches the top two current firefox User-Agents:

```
Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0
Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/116.0
```
source: https://techblog.willshouse.com/2012/01/03/most-common-user-agents/ and my browser
```
Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:117.0) Gecko/20100101 Firefox/117.0
```
This doesn't exist in the wild^ hence we get detected as a bot.

Reply:
Actually that's good catch.
I confirm my user agent is `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/117.0`

Some context:
* https://bugzilla.mozilla.org/show_bug.cgi?id=1805967
* https://hg.mozilla.org/mozilla-central/rev/c7d69ab92660

Reply:
Wow .. can't believe: ["cap rv: bits in User Agent string to 109 because some sites detect IE11 based on rv:11*,"](https://hg.mozilla.org/mozilla-central/rev/c7d69ab92660) .. @jazzzooo good catch :+1:

What do you think .. will they stick the `rv` at 109 forever or just until there is a FFox version 120?

If I understand it right, they do not stick the Android version: ["keep android the same because the issue doesn't occur there"](https://hg.mozilla.org/mozilla-central/rev/a186000b3a89)

Since we do not know if they stick at 109 forever and to be future proof, should we change from to a [Android User-Agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent/Firefox#android_version_41_and_above)?

Reply:
> should we change from to a Android User-Agent?

I don't think so: the search engines will return the mobile version of the web sites --> if we do that, all the existing engines have to be reviewed and some updated.

We just have to be careful at the version 120: we break on purpose update_firefox_version.py on the version 120 to make sure we check the value.